### PR TITLE
Fix(UI): Regression in search filters

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on #9108 
Added unit tests for newly added functionality

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">

Regression in ES facet filters:

https://user-images.githubusercontent.com/51777795/206423902-8392b618-1d25-4c16-a41f-0c2e02692328.mov

Combined results for ES and Advanced Search filters:

https://user-images.githubusercontent.com/51777795/206424053-86471d3a-91fb-4935-a241-675f6863acdb.mov

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/enums/Explore.enum.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export enum QueryFilterFieldsEnum {
+  MUST = 'must',
+  SHOULD = 'should',
+}
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/enums/Explore.enum.ts:4]: licance

File: openmetadata-ui/src/main/resources/ui/src/pages/explore/ExplorePage.component.tsx [hunk 2]
```diff
@@ -191,8 +193,10 @@ const ExplorePage: FunctionComponent = () => {
       searchQuery({
         query: searchQueryParam,
         searchIndex,
-        queryFilter: advancesSearchQueryFilter,
-        postFilter: elasticsearchPostFilterQuery,
+        queryFilter: getCombinedQueryFilterObject(
+          elasticsearchQueryFilter as unknown as QueryFilterInterface,
+          advancesSearchQueryFilter as unknown as QueryFilterInterface
+        ),
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/explore/ExplorePage.component.tsx:199]: Why are we typecasting?

Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/explore/ExplorePage.component.tsx:199]: Both query filter objects have type as `Record<string, unknown>`, so here `unknown` will not allow us to directly access the properties, that's why we have to first typecast it into some type and then we can access the properties.
I will add the comment in the code as well.

File: openmetadata-ui/src/main/resources/ui/src/enums/Explore.enum.ts (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/enums/Explore.enum.ts:17]: Added

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**